code/modules/antagonists/voidwalker/voidwalker_kidnap.dm

```diff
@@ -0,0 +1,173 @@
+
+/datum/component/space_kidnap/proc/kidnap(mob/living/parent, mob/living/victim)
+	victim.Paralyze(kidnap_time) //so they don't get up if we already got em
+	var/obj/particles = new /obj/effect/abstract/particle_holder (victim, /particles/void_kidnap)
+	kidnapping = TRUE
+
+	if(do_after(parent, kidnap_time, victim, extra_checks = CALLBACK(victim, TYPE_PROC_REF(/mob, incapacitated))))
+		take_them(victim)
+
+	qdel(particles)
+	kidnapping = FALSE
+
+/datum/component/space_kidnap/proc/take_them(mob/living/victim)
+	if(ishuman(victim))
+		var/mob/living/carbon/human/hewmon = victim
+		hewmon.gain_trauma(/datum/brain_trauma/voided)
+
+	victim.flash_act(INFINITY, override_blindness_check = TRUE, visual = TRUE, type = /atom/movable/screen/fullscreen/flash/black)
+	new /obj/effect/temp_visual/circle_wave/unsettle(get_turf(victim))
+
+	if(!SSmapping.lazy_load_template(LAZY_TEMPLATE_KEY_VOIDWALKER_VOID) || !GLOB.voidwalker_void.len)
+		dump(victim)
+		victim.heal_overall_damage(brute = 80, burn = 20)
+		CRASH("[victim] was instantly dumped after being voidwalker kidnapped due to a missing landmark!")
+	else
+		victim.heal_and_revive(90)
+
+		victim.adjustOxyLoss(-100, FALSE)
+
+		new /obj/effect/wisp_mobile (get_turf(pick(GLOB.voidwalker_void)), victim)
+		addtimer(CALLBACK(src, PROC_REF(dump), victim), 60 SECONDS)
+
+/datum/component/space_kidnap/proc/dump(mob/living/trash)
+	trash.forceMove(get_random_station_turf())
+
+/// A global assoc list for the drop of point
+GLOBAL_LIST_EMPTY(voidwalker_void)
+
+/// Lardmarks meant to designate where voidwalker kidnapees are sent
+/obj/effect/landmark/voidwalker_void
+	name = "default voidwalker void landmark"
+	icon_state = "x"
+
+/obj/effect/landmark/voidwalker_void/Initialize(mapload)
+	. = ..()
+	GLOB.voidwalker_void += src
+
+/obj/effect/landmark/voidwalker_void/Destroy()
+	GLOB.voidwalker_void -= src
+	return ..()
+
+/// Voidwalker void where the people go
+/area/centcom/voidwalker_void
+	name = "Voidwalker void"
+	icon_state = "voidwalker"
+	has_gravity = STANDARD_GRAVITY
+	ambience_index = AMBIENCE_SPOOKY
+	sound_environment = SOUND_ENVIRONMENT_CAVE
+	area_flags = UNIQUE_AREA | NOTELEPORT | HIDDEN_AREA | BLOCK_SUICIDE
+
+/// Mini car where people drive around in in their mangled corpse to heal a bit before they get dumped back on station
+/obj/effect/wisp_mobile
+	name = "wisp"
+
+	icon = 'icons/obj/weapons/voidwalker_items.dmi'
+	icon_state = "wisp"
+
+	light_system = OVERLAY_LIGHT
+	light_color = COLOR_WHITE
+	light_range = 4
+	light_power = 1
+	light_on = TRUE
+
+	/// Delay between movements
+	var/move_delay = 0.5 SECONDS
+	/// when can we move again?
+	var/can_move
+	/// what do we eatt?
+	var/food_type = /obj/effect/wisp_food
+	/// how much do we heal per food?
+	var/heal_per_food = 15
+
+/obj/effect/wisp_mobile/Initialize(mapload, mob/living/driver)
+	. = ..()
+
+	if(isliving(driver))
+		driver.forceMove(src)
+		driver.add_traits(list(TRAIT_STASIS, TRAIT_NOSOFTCRIT, TRAIT_NOHARDCRIT), REF(src))
```

Comment:
Also same story about "why not make this list a static property of the effect type?"